lib/mongo/database.rb

```diff
@@ -126,12 +126,16 @@ def collection_names(options = {})
     #   information hash per collection, depends on the MongoDB server
     #   version that fulfills the request.
     #
+    # @param [ Hash ] filter A filter on the collections returned by
+    #   this command. See https://docs.mongodb.com/manual/reference/command/listCollections/
+    #   for documentation on the filter option.
+    #
     # @return [ Array<Hash> ] Array of information hashes, one for each
     #   collection in the database.
     #
     # @since 2.0.5
-    def list_collections
-      View.new(self).list_collections
+    def list_collections(filter={})
```

Comment:
@p-mongo I'm wondering if it would make more sense to make this a double-splat argument as well, since we're going to eventually want to change this method to take multiple options.

Reply:
Yes please.